File: backend/ContentGenerator.Domain/Entities/Project.cs
namespace ContentGenerator.Domain.Entities;

public enum ProjectStatus
{
    Draft,
    GeneratingScript,
    UserReview,
    GeneratingAudio,
    GeneratingVideo,
    Completed,
    Failed
}

public class Project
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    
    public ProjectStatus Status { get; set; } = ProjectStatus.Draft;
    
    // One-to-One
    public VideoSettings Settings { get; set; } = null!;
    public ScriptLayer? Script { get; set; }
    
    // One-to-Many
    public ICollection<MediaLayer> MediaLayers { get; set; } = new List<MediaLayer>();

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/app/types/project.ts
export enum ProjectStatus {
  Draft = 0,
  GeneratingScript = 1,
  UserReview = 2,
  GeneratingAudio = 3,
  GeneratingVideo = 4,
  Completed = 5
}

export interface Project {
  id: string;
  title: string;
  description?: string;
  status: ProjectStatus;
  generatedScript?: string;
  audioUrl?: string;
  videoUrl?: string;
  _cached?: boolean; // temporary flag for UI
}

export interface User {
  id: string;
  email: string;
}
